From 678ac9657d3166dcdeadbccb7ad9427ba0677339 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Thu, 26 Feb 2015 07:52:30 -0500
Subject: [PATCH 159/315] display: turn some assertions into warnings

It's clear there's a bug if they're hit, but
we shouldn't tank, since it's not a fatal
bug.
---
 daemon/gdm-display.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/daemon/gdm-display.c b/daemon/gdm-display.c
index 1d20284f..08b6ef4b 100644
--- a/daemon/gdm-display.c
+++ b/daemon/gdm-display.c
@@ -1215,10 +1215,10 @@ gdm_display_dispose (GObject *object)
         }
         g_clear_object (&self->priv->launch_environment);
 
-        g_assert (self->priv->status == GDM_DISPLAY_FINISHED ||
-                  self->priv->status == GDM_DISPLAY_FAILED);
-        g_assert (self->priv->user_access_file == NULL);
-        g_assert (self->priv->access_file == NULL);
+        g_warn_if_fail (self->priv->status == GDM_DISPLAY_PREPARED);
+        g_warn_if_fail (self->priv->status == GDM_DISPLAY_MANAGED);
+        g_warn_if_fail (self->priv->user_access_file == NULL);
+        g_warn_if_fail (self->priv->access_file == NULL);
 
         G_OBJECT_CLASS (gdm_display_parent_class)->dispose (object);
 }
-- 
2.15.1

